Updated README (#8)
cosmoscout/docker-mapserver: README.md

<!-- 
SPDX-FileCopyrightText: German Aerospace Center (DLR) <cosmoscout@dlr.de>
SPDX-License-Identifier: CC0-1.0
 -->
# Map-Server Container for CosmoScout VR

A dockerized mapserver for CosmoScout [csp-lod bodies](https://github.com/cosmoscout/cosmoscout-vr/tree/main/plugins/csp-lod-bodies#readme). This repo has been created taking three use-cases into consideration - `loading example map data`, `loading custom map data` and `creating custom map data containers`. Each of the use-cases are discussed briefly below.

## 1) Loading Example Map Data

> Here you run a dockerized mapserver using NASA's `Blue Marble` and the `Natural Earth image` datasets and `ETOPO1` elevation data.

Reviewer: Would be cool if you could turn the dataset names into links to websites containing more information on the data.

### Running the Server
To run the dockerized mapserver, you can pull the docker image (built from the `example.Dockerfile`) from GHCR using the command:

```bash
docker run -ti --rm -p 8080:80 ghcr.io/cosmoscout/mapserver-example
```
The command binds localhost port 8080 to container port 80.

> To ensure that everything is working as intended, see if you can access the example-datasets using the following links:

Reviewer: I guess that it would be more helpful if the links below were not in a code  block but actual clickable links.

# One base patch.
http://localhost:8080/cgi-bin/mapserv?map=/mapserver-datasets/meta.map&service=wms&version=1.3.0&request=GetMap&layers=earth.naturalearth.rgb&bbox=3,2,4,3&width=800&height=800&crs=epsg:900914&format=pngRGB
```
### Configuring CosmoScout VR:

> Now that the datasets are working, we only need to include them into CosmoScout VR. To do this, add the following section to the "plugins" array in your `"share/config/simple_desktop.json"`. You will have to adjust the mapserver links according to the location of your meta.map file.

Reviewer: ```suggestion
Now that the datasets are working, we only need to include them into CosmoScout VR. To do this, add the following section to the "plugins" object in your `"share/config/simple_desktop.json"`. 
```

Also, the links in the configuration below are missing the :8080, right?

},
...
```
You should also remove the "Earth" section from the "csp-simple-bodies" plugin configuration in the same file, else you will have two Earths drawn on top of each other!

## 2) Loading Custom Map Data
Here you can run a dockerized mapserver using your own custom dataset.

### Preparing the Map Data

> As you are using your own custom dataset some changes need to be made to the `meta.map` file in the `mapserver-datasets` directory. The lines `42,43 and 44` need to change accordingly to the location of your `.map` files.

Reviewer: Please describe in more detail how to set up the data directory. Imagine you have a tiff file for some location on Earth, which steps are necessary to serve this with the mapserver? This definitely involves:
* Setting up a mapserver-datasets directory with an appropriate directory structure inside.
* Copying the epsg and meta.map files from the repository
*  Adding a custom map file for the tif
* Maybe optimizing the tif with `gdal_translate` and `gdaladdo`
* Running the container
* Finally configuring CosmoScout VR
